packages/bot/src/scratch/index.js

```diff
@@ -74,15 +74,24 @@ export const scratchWorkspaceInit = async () => {
 let interpreter = null;
 
 export const runBot = (limitations = {}) => {
-    // Disable blocks outside of any main or independent blocks.
-    const top_blocks = Blockly.derivWorkspace.getTopBlocks();
-    top_blocks.forEach(block => {
-        if (!block.isMainBlock() && !block.isIndependentBlock()) {
-            block.setDisabled(true);
+    return new Promise((resolve, reject) => {
```

Comment:
- Instead of creating a `runBot` promise we could do a sanity check before attempting to run the strategy, i.e. by creating a `checkStrategy` function outside `runBot` that ensures the XML correct and valid to be executed. This will save us from promise-hell. 
- You could pass an optional callback to `runBot` that would be executed if the bot is started running.